### ISSUE TITLE
e2e SaveLogs overloads the build-log, and corrupts the junit report.

### ISSUE THREAD

**Issue by therealmitchconnors** [istio/istio#15332]
(NOTE: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities/>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>
)

**Bug description**
When an e2e test fails, we save tons of cluster info, which is great, but we save it to stdout in a non-parseable format (https://github.com/istio/istio/blob/master/tests/e2e/framework/framework.go#L217).  This is not challenging to read, challenging to download (build-log.txt gets up to 4 GB), and corrupts our junit-report output (which parses build-log.txt expecting go test like output).  See https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/12276/e2e-mixer-no_auth-master/3174/build-log.txt for an example of a huge build log, and https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/12276/e2e-mixer-no_auth-master/3174/artifacts/junit.xml for the corresponding junit.xml.

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ x ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
junit.xml contains actual test case information, build-log is manageable, k8s logs are stored separately.

**Environment where bug was observed (cloud vendor, OS, etc)**

prow.istio.io

**Comment by howardjohn** [comment]
I think we already store the logs as files but the `kubectl describe` dump we do is to stdout, so we can just move that to files?

Big +1 on this though, I do not like the huge describe dump

**Comment by therealmitchconnors** [comment]
Good point.  I propose that we store as both yaml and text for ease of use.  

Also at, https://github.com/istio/istio/blob/master/tests/e2e/framework/testInfo.go#L69 can I drop the ID from the folder name?  K8S workloads should have isolated file systems, and this ID makes the gcs bucket difficult (but not impossible) to read.

**Comment by kyessenov** [comment]
I was thinking whether log dumping is necessary for prow on GKE. All the data is already stored in GCP monitoring, so we just need to create the right queries to provide access to the logs, as well as an authorization proxy.

**Comment by howardjohn** [comment]
We will probably/possibly move our tests to running in KinD so I am not sure we should rely on GCP monitoring

**Comment by kyessenov** [comment]
I'm not saying we should disallow fetching logs directly. It's just going to be a lot faster not to fetch logs on GKE. I think up to half of the time was spent reading through logs (we can also log less of course).

**Comment by howardjohn** [comment]
Ah I see what you mean. Makes sense to me if it works

**Comment by therealmitchconnors** [comment]
As John pointed out, logs are not impacting the build-log.  The real problem is the describe command, which I am fixing now.  If changes to how logs are fetched are needed, they should be addressed in a separate issue.

**Comment by kyessenov** [comment]
The test framework is not fetching logs on failure which is why there is no impact yet. I agree this is a separate issue, just wanted to make sure we're aware of the GKE monitoring option.
